Feature detector nomenclature

wdconinc opened the pull request:
This implements the detector nomenclature scheme.

`perl cadGeneratorV1.pl -F cadp.csv` now produces the following detector number and respective angle. Assumption is 0 deg corresponds with closed sector of coil A, start of septant 1.
```
ring 1
101010 at 0.00 ( = 0.00 + 0.00 ) deg
110110 at 12.86 ( = 0.00 + 12.86 ) deg
110210 at 25.71 ( = 0.00 + 25.71 ) deg
110310 at 38.57 ( = 0.00 + 38.57 ) deg
102010 at 51.43 ( = 51.43 + 0.00 ) deg
120110 at 64.29 ( = 51.43 + 12.86 ) deg
120210 at 77.14 ( = 51.43 + 25.71 ) deg
120310 at 90.00 ( = 51.43 + 38.57 ) deg
103010 at 102.86 ( = 102.86 + 0.00 ) deg
130110 at 115.71 ( = 102.86 + 12.86 ) deg
130210 at 128.57 ( = 102.86 + 25.71 ) deg
130310 at 141.43 ( = 102.86 + 38.57 ) deg
104010 at 154.29 ( = 154.29 + 0.00 ) deg
140110 at 167.14 ( = 154.29 + 12.86 ) deg
140210 at 180.00 ( = 154.29 + 25.71 ) deg
140310 at 192.86 ( = 154.29 + 38.57 ) deg
105010 at 205.71 ( = 205.71 + 0.00 ) deg
150110 at 218.57 ( = 205.71 + 12.86 ) deg
150210 at 231.43 ( = 205.71 + 25.71 ) deg
150310 at 244.29 ( = 205.71 + 38.57 ) deg
106010 at 257.14 ( = 257.14 + 0.00 ) deg
160110 at 270.00 ( = 257.14 + 12.86 ) deg
160210 at 282.86 ( = 257.14 + 25.71 ) deg
160310 at 295.71 ( = 257.14 + 38.57 ) deg
107010 at 308.57 ( = 308.57 + 0.00 ) deg
170110 at 321.43 ( = 308.57 + 12.86 ) deg
170210 at 334.29 ( = 308.57 + 25.71 ) deg
170310 at 347.14 ( = 308.57 + 38.57 ) deg
ring 2
201010 at 0.00 ( = 0.00 + 0.00 ) deg
210110 at 12.86 ( = 0.00 + 12.86 ) deg
210210 at 25.71 ( = 0.00 + 25.71 ) deg
210310 at 38.57 ( = 0.00 + 38.57 ) deg
202010 at 51.43 ( = 51.43 + 0.00 ) deg
220110 at 64.29 ( = 51.43 + 12.86 ) deg
220210 at 77.14 ( = 51.43 + 25.71 ) deg
220310 at 90.00 ( = 51.43 + 38.57 ) deg
203010 at 102.86 ( = 102.86 + 0.00 ) deg
230110 at 115.71 ( = 102.86 + 12.86 ) deg
230210 at 128.57 ( = 102.86 + 25.71 ) deg
230310 at 141.43 ( = 102.86 + 38.57 ) deg
204010 at 154.29 ( = 154.29 + 0.00 ) deg
240110 at 167.14 ( = 154.29 + 12.86 ) deg
240210 at 180.00 ( = 154.29 + 25.71 ) deg
240310 at 192.86 ( = 154.29 + 38.57 ) deg
205010 at 205.71 ( = 205.71 + 0.00 ) deg
250110 at 218.57 ( = 205.71 + 12.86 ) deg
250210 at 231.43 ( = 205.71 + 25.71 ) deg
250310 at 244.29 ( = 205.71 + 38.57 ) deg
206010 at 257.14 ( = 257.14 + 0.00 ) deg
260110 at 270.00 ( = 257.14 + 12.86 ) deg
260210 at 282.86 ( = 257.14 + 25.71 ) deg
260310 at 295.71 ( = 257.14 + 38.57 ) deg
207010 at 308.57 ( = 308.57 + 0.00 ) deg
270110 at 321.43 ( = 308.57 + 12.86 ) deg
270210 at 334.29 ( = 308.57 + 25.71 ) deg
270310 at 347.14 ( = 308.57 + 38.57 ) deg
ring 3
301010 at 0.00 ( = 0.00 + 0.00 ) deg
310110 at 12.86 ( = 0.00 + 12.86 ) deg
310210 at 25.71 ( = 0.00 + 25.71 ) deg
310310 at 38.57 ( = 0.00 + 38.57 ) deg
302010 at 51.43 ( = 51.43 + 0.00 ) deg
320110 at 64.29 ( = 51.43 + 12.86 ) deg
320210 at 77.14 ( = 51.43 + 25.71 ) deg
320310 at 90.00 ( = 51.43 + 38.57 ) deg
303010 at 102.86 ( = 102.86 + 0.00 ) deg
330110 at 115.71 ( = 102.86 + 12.86 ) deg
330210 at 128.57 ( = 102.86 + 25.71 ) deg
330310 at 141.43 ( = 102.86 + 38.57 ) deg
304010 at 154.29 ( = 154.29 + 0.00 ) deg
340110 at 167.14 ( = 154.29 + 12.86 ) deg
340210 at 180.00 ( = 154.29 + 25.71 ) deg
340310 at 192.86 ( = 154.29 + 38.57 ) deg
305010 at 205.71 ( = 205.71 + 0.00 ) deg
350110 at 218.57 ( = 205.71 + 12.86 ) deg
350210 at 231.43 ( = 205.71 + 25.71 ) deg
350310 at 244.29 ( = 205.71 + 38.57 ) deg
306010 at 257.14 ( = 257.14 + 0.00 ) deg
360110 at 270.00 ( = 257.14 + 12.86 ) deg
360210 at 282.86 ( = 257.14 + 25.71 ) deg
360310 at 295.71 ( = 257.14 + 38.57 ) deg
307010 at 308.57 ( = 308.57 + 0.00 ) deg
370110 at 321.43 ( = 308.57 + 12.86 ) deg
370210 at 334.29 ( = 308.57 + 25.71 ) deg
370310 at 347.14 ( = 308.57 + 38.57 ) deg
ring 4
401010 at 0.00 ( = 0.00 + 0.00 ) deg
410110 at 12.86 ( = 0.00 + 12.86 ) deg
410210 at 25.71 ( = 0.00 + 25.71 ) deg
410310 at 38.57 ( = 0.00 + 38.57 ) deg
402010 at 51.43 ( = 51.43 + 0.00 ) deg
420110 at 64.29 ( = 51.43 + 12.86 ) deg
420210 at 77.14 ( = 51.43 + 25.71 ) deg
420310 at 90.00 ( = 51.43 + 38.57 ) deg
403010 at 102.86 ( = 102.86 + 0.00 ) deg
430110 at 115.71 ( = 102.86 + 12.86 ) deg
430210 at 128.57 ( = 102.86 + 25.71 ) deg
430310 at 141.43 ( = 102.86 + 38.57 ) deg
404010 at 154.29 ( = 154.29 + 0.00 ) deg
440110 at 167.14 ( = 154.29 + 12.86 ) deg
440210 at 180.00 ( = 154.29 + 25.71 ) deg
440310 at 192.86 ( = 154.29 + 38.57 ) deg
405010 at 205.71 ( = 205.71 + 0.00 ) deg
450110 at 218.57 ( = 205.71 + 12.86 ) deg
450210 at 231.43 ( = 205.71 + 25.71 ) deg
450310 at 244.29 ( = 205.71 + 38.57 ) deg
406010 at 257.14 ( = 257.14 + 0.00 ) deg
460110 at 270.00 ( = 257.14 + 12.86 ) deg
460210 at 282.86 ( = 257.14 + 25.71 ) deg
460310 at 295.71 ( = 257.14 + 38.57 ) deg
407010 at 308.57 ( = 308.57 + 0.00 ) deg
470110 at 321.43 ( = 308.57 + 12.86 ) deg
470210 at 334.29 ( = 308.57 + 25.71 ) deg
470310 at 347.14 ( = 308.57 + 38.57 ) deg
ring 5.0
510200 at 21.43 ( = 0.00 + 21.43 ) deg
510210 at 25.71 ( = 0.00 + 25.71 ) deg
510220 at 30.00 ( = 0.00 + 30.00 ) deg
520200 at 72.86 ( = 51.43 + 21.43 ) deg
520210 at 77.14 ( = 51.43 + 25.71 ) deg
520220 at 81.43 ( = 51.43 + 30.00 ) deg
530200 at 124.29 ( = 102.86 + 21.43 ) deg
530210 at 128.57 ( = 102.86 + 25.71 ) deg
530220 at 132.86 ( = 102.86 + 30.00 ) deg
540200 at 175.71 ( = 154.29 + 21.43 ) deg
540210 at 180.00 ( = 154.29 + 25.71 ) deg
540220 at 184.29 ( = 154.29 + 30.00 ) deg
550200 at 227.14 ( = 205.71 + 21.43 ) deg
550210 at 231.43 ( = 205.71 + 25.71 ) deg
550220 at 235.71 ( = 205.71 + 30.00 ) deg
560200 at 278.57 ( = 257.14 + 21.43 ) deg
560210 at 282.86 ( = 257.14 + 25.71 ) deg
560220 at 287.14 ( = 257.14 + 30.00 ) deg
570200 at 330.00 ( = 308.57 + 21.43 ) deg
570210 at 334.29 ( = 308.57 + 25.71 ) deg
570220 at 338.57 ( = 308.57 + 30.00 ) deg
ring 5.1
510100 at 8.57 ( = 0.00 + 8.57 ) deg
510110 at 12.86 ( = 0.00 + 12.86 ) deg
510120 at 17.14 ( = 0.00 + 17.14 ) deg
510100 at 34.29 ( = 0.00 + 34.29 ) deg
510110 at 38.57 ( = 0.00 + 38.57 ) deg
510120 at 42.86 ( = 0.00 + 42.86 ) deg
520100 at 60.00 ( = 51.43 + 8.57 ) deg
520110 at 64.29 ( = 51.43 + 12.86 ) deg
520120 at 68.57 ( = 51.43 + 17.14 ) deg
520100 at 85.71 ( = 51.43 + 34.29 ) deg
520110 at 90.00 ( = 51.43 + 38.57 ) deg
520120 at 94.29 ( = 51.43 + 42.86 ) deg
530100 at 111.43 ( = 102.86 + 8.57 ) deg
530110 at 115.71 ( = 102.86 + 12.86 ) deg
530120 at 120.00 ( = 102.86 + 17.14 ) deg
530100 at 137.14 ( = 102.86 + 34.29 ) deg
530110 at 141.43 ( = 102.86 + 38.57 ) deg
530120 at 145.71 ( = 102.86 + 42.86 ) deg
540100 at 162.86 ( = 154.29 + 8.57 ) deg
540110 at 167.14 ( = 154.29 + 12.86 ) deg
540120 at 171.43 ( = 154.29 + 17.14 ) deg
540100 at 188.57 ( = 154.29 + 34.29 ) deg
540110 at 192.86 ( = 154.29 + 38.57 ) deg
540120 at 197.14 ( = 154.29 + 42.86 ) deg
550100 at 214.29 ( = 205.71 + 8.57 ) deg
550110 at 218.57 ( = 205.71 + 12.86 ) deg
550120 at 222.86 ( = 205.71 + 17.14 ) deg
550100 at 240.00 ( = 205.71 + 34.29 ) deg
550110 at 244.29 ( = 205.71 + 38.57 ) deg
550120 at 248.57 ( = 205.71 + 42.86 ) deg
560100 at 265.71 ( = 257.14 + 8.57 ) deg
560110 at 270.00 ( = 257.14 + 12.86 ) deg
560120 at 274.29 ( = 257.14 + 17.14 ) deg
560100 at 291.43 ( = 257.14 + 34.29 ) deg
560110 at 295.71 ( = 257.14 + 38.57 ) deg
560120 at 300.00 ( = 257.14 + 42.86 ) deg
570100 at 317.14 ( = 308.57 + 8.57 ) deg
570110 at 321.43 ( = 308.57 + 12.86 ) deg
570120 at 325.71 ( = 308.57 + 17.14 ) deg
570100 at 342.86 ( = 308.57 + 34.29 ) deg
570110 at 347.14 ( = 308.57 + 38.57 ) deg
570120 at 351.43 ( = 308.57 + 42.86 ) deg
ring 5.2
501000 at -4.29 ( = 0.00 + -4.29 ) deg
501010 at 0.00 ( = 0.00 + 0.00 ) deg
501020 at 4.29 ( = 0.00 + 4.29 ) deg
502000 at 47.14 ( = 51.43 + 47.14 ) deg
502010 at 51.43 ( = 51.43 + 0.00 ) deg
502020 at 55.71 ( = 51.43 + 4.29 ) deg
503000 at 98.57 ( = 102.86 + 47.14 ) deg
503010 at 102.86 ( = 102.86 + 0.00 ) deg
503020 at 107.14 ( = 102.86 + 4.29 ) deg
504000 at 150.00 ( = 154.29 + 47.14 ) deg
504010 at 154.29 ( = 154.29 + 0.00 ) deg
504020 at 158.57 ( = 154.29 + 4.29 ) deg
505000 at 201.43 ( = 205.71 + 47.14 ) deg
505010 at 205.71 ( = 205.71 + 0.00 ) deg
505020 at 210.00 ( = 205.71 + 4.29 ) deg
506000 at 252.86 ( = 257.14 + 47.14 ) deg
506010 at 257.14 ( = 257.14 + 0.00 ) deg
506020 at 261.43 ( = 257.14 + 4.29 ) deg
507000 at 304.29 ( = 308.57 + 47.14 ) deg
507010 at 308.57 ( = 308.57 + 0.00 ) deg
507020 at 312.86 ( = 308.57 + 4.29 ) deg
ring 6
601010 at 0.00 ( = 0.00 + 0.00 ) deg
610110 at 12.86 ( = 0.00 + 12.86 ) deg
610210 at 25.71 ( = 0.00 + 25.71 ) deg
610310 at 38.57 ( = 0.00 + 38.57 ) deg
602010 at 51.43 ( = 51.43 + 0.00 ) deg
620110 at 64.29 ( = 51.43 + 12.86 ) deg
620210 at 77.14 ( = 51.43 + 25.71 ) deg
620310 at 90.00 ( = 51.43 + 38.57 ) deg
603010 at 102.86 ( = 102.86 + 0.00 ) deg
630110 at 115.71 ( = 102.86 + 12.86 ) deg
630210 at 128.57 ( = 102.86 + 25.71 ) deg
630310 at 141.43 ( = 102.86 + 38.57 ) deg
604010 at 154.29 ( = 154.29 + 0.00 ) deg
640110 at 167.14 ( = 154.29 + 12.86 ) deg
640210 at 180.00 ( = 154.29 + 25.71 ) deg
640310 at 192.86 ( = 154.29 + 38.57 ) deg
605010 at 205.71 ( = 205.71 + 0.00 ) deg
650110 at 218.57 ( = 205.71 + 12.86 ) deg
650210 at 231.43 ( = 205.71 + 25.71 ) deg
650310 at 244.29 ( = 205.71 + 38.57 ) deg
606010 at 257.14 ( = 257.14 + 0.00 ) deg
660110 at 270.00 ( = 257.14 + 12.86 ) deg
660210 at 282.86 ( = 257.14 + 25.71 ) deg
660310 at 295.71 ( = 257.14 + 38.57 ) deg
607010 at 308.57 ( = 308.57 + 0.00 ) deg
670110 at 321.43 ( = 308.57 + 12.86 ) deg
670210 at 334.29 ( = 308.57 + 25.71 ) deg
670310 at 347.14 ( = 308.57 + 38.57 ) deg
```

wdconinc committed:
Write the correct DetNo for detector parts per nomenclature

PMT Cathode (0), Quartz (1), Mirror (2), Reflector Skin (3),
Reflector Volume (4), Lightguide Skin (5), Lightguide Volume (6),
PMT Housing (7)

wdconinc committed:
perltidy and use correct nomenclature for DetNo.

wdconinc commented:
Note that I was too lazy to fix:
```
ring 5.2
501000 at -4.29 ( = 0.00 + -4.29 ) deg
501010 at 0.00 ( = 0.00 + 0.00 ) deg
501020 at 4.29 ( = 0.00 + 4.29 ) deg
502000 at 47.14 ( = 51.43 + 47.14 ) deg
502010 at 51.43 ( = 51.43 + 0.00 ) deg
502020 at 55.71 ( = 51.43 + 4.29 ) deg
```
figuring that it was obvious what is happening with the rounding there.

wdconinc committed:
Now also do the right thing for 5.1

wdconinc commented:
Ring 5, transition didn't have a separation between trans- and trans+.

```
ring 5.1
510100 at 8.57 ( = 0.00 + 8.57 ) deg
510110 at 12.86 ( = 0.00 + 12.86 ) deg
510120 at 17.14 ( = 0.00 + 17.14 ) deg
510300 at 34.29 ( = 0.00 + 34.29 ) deg
510310 at 38.57 ( = 0.00 + 38.57 ) deg
510320 at 42.86 ( = 0.00 + 42.86 ) deg
520100 at 60.00 ( = 51.43 + 8.57 ) deg
520110 at 64.29 ( = 51.43 + 12.86 ) deg
520120 at 68.57 ( = 51.43 + 17.14 ) deg
520300 at 85.71 ( = 51.43 + 34.29 ) deg
520310 at 90.00 ( = 51.43 + 38.57 ) deg
520320 at 94.29 ( = 51.43 + 42.86 ) deg
530100 at 111.43 ( = 102.86 + 8.57 ) deg
530110 at 115.71 ( = 102.86 + 12.86 ) deg
530120 at 120.00 ( = 102.86 + 17.14 ) deg
530300 at 137.14 ( = 102.86 + 34.29 ) deg
530310 at 141.43 ( = 102.86 + 38.57 ) deg
530320 at 145.71 ( = 102.86 + 42.86 ) deg
540100 at 162.86 ( = 154.29 + 8.57 ) deg
540110 at 167.14 ( = 154.29 + 12.86 ) deg
540120 at 171.43 ( = 154.29 + 17.14 ) deg
540300 at 188.57 ( = 154.29 + 34.29 ) deg
540310 at 192.86 ( = 154.29 + 38.57 ) deg
540320 at 197.14 ( = 154.29 + 42.86 ) deg
550100 at 214.29 ( = 205.71 + 8.57 ) deg
550110 at 218.57 ( = 205.71 + 12.86 ) deg
550120 at 222.86 ( = 205.71 + 17.14 ) deg
550300 at 240.00 ( = 205.71 + 34.29 ) deg
550310 at 244.29 ( = 205.71 + 38.57 ) deg
550320 at 248.57 ( = 205.71 + 42.86 ) deg
560100 at 265.71 ( = 257.14 + 8.57 ) deg
560110 at 270.00 ( = 257.14 + 12.86 ) deg
560120 at 274.29 ( = 257.14 + 17.14 ) deg
560300 at 291.43 ( = 257.14 + 34.29 ) deg
560310 at 295.71 ( = 257.14 + 38.57 ) deg
560320 at 300.00 ( = 257.14 + 42.86 ) deg
570100 at 317.14 ( = 308.57 + 8.57 ) deg
570110 at 321.43 ( = 308.57 + 12.86 ) deg
570120 at 325.71 ( = 308.57 + 17.14 ) deg
570300 at 342.86 ( = 308.57 + 34.29 ) deg
570310 at 347.14 ( = 308.57 + 38.57 ) deg
570320 at 351.43 ( = 308.57 + 42.86 ) deg
```

wdconinc committed:
perltidy.

wdconinc committed:
Implemented new detector nomenclature in detector

wdconinc committed:
Bug fix, missing dollar sign

wdconinc committed:
Add updated numbering scheme

wdconinc committed:
Moved unused files to deprecated directory

wdconinc committed:
Remove constant PI

wdconinc committed:
Apply same detno modifications to inline PMTs

wdconinc committed:
Integrated inline PMT option by passing -I with non-empty string

wdconinc committed:
Removed redundant inlinePMTs file

wdconinc committed:
Added line on -I option

cameronc137 commented:
I'll be using this branch soon to make some detector tests. If I find any issues I'll make more commits here.

cameronc137 commented:
I get a bug on ifarm, my desktop, and another computer about a missing POSIX::Round function, and I after a bit of effort I still can't figure out how to install it or make the error message go away from my perl install's end.

wdconinc commented:
Ok, I'll remove the round functionality. I guess we can't rely on having
posix functionality...

Cheers,
Wouter

On Sat, Jul 4, 2020, 10:46 PM Cameron Clarke <notifications@github.com>
wrote:

> I get a bug on ifarm, my desktop, and another computer about a missing
> POSIX::Round function, and I after a bit of effort I still can't figure out
> how to install it or make the error message go away from my perl install's
> end.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/JeffersonLab/remoll-detector-generator/pull/13#issuecomment-653838422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDQ2B5PVBO3ASHHV7B6S4TRZ7ZRTANCNFSM4MWVISHQ>
> .
>

wdconinc committed:
POSIX::round was only introduced with perl 5.22, and ifarm has 5.20

This introduces a proper rounding that handles both positive and
negative numbers well, and that does the right (POSIX) thing at the
halfway point.

Fortunately POSIX::fmod and POSIX::modf are available in 5.20 (but
not available by default).

wdconinc commented:
Fixed with d48c9f9.

cameronc137 commented on gdmlGeneratorV1_materials.pl:
```diff
@@ -437,202 +476,342 @@
 	<second ref=\"coneMotherSol${opt_T}\"/> 
 	<position unit=\"mm\" name=\"coneMotherPos${opt_T}\" x=\"0\" y=\"0\" z=\"0\"\/>
-        <rotation unit=\"rad\" name=\"coneMotherRot${opt_T}\" x=\"0\" y=\"PI/2\" z=\"0\"/>
+        <rotation unit=\"rad\" name=\"coneMotherRot${opt_T}\" x=\"0\" y=\"pi/2\" z=\"0\"/>
```
Should all of these PI's be lowercase now?

wdconinc replied:
I don't know. Doesn't it work?

cameronc137 replied:
Yes, it is working

cameronc137 commented on gdmlGeneratorV1_materials.pl:
```diff
@@ -9,224 +9,281 @@
+        print
+          "String contains 0 or more white-space character and nothing else.\n";
+    }
+    else {
```
I think these ring 5 checks are lacking the 6 digit nomenclature

cameronc137 replied:
I will have a commit to fix this momentarily

cameronc137 commented on gdmlGeneratorV1_materials.pl:
```diff
@@ -9,224 +9,281 @@
+                && substr( trim( $fields[0] ), 0, 5 ) == "52002" )
+            {
+            }
+            elsif (
```
I think this piece of logic was trying to include all 3 sectors if the words open, trans or closed weren't included at all.

cameronc137 committed:
Updating gdmlGenerator to have ring 5 numbers,
and correcting the optical properties in materialsOptical.xml,
and getting the geometry to better match the qsim benchmark
that it was initially designed to imitate (though nuances of
PMT cathode optical properties and reflector segment reflectivity
will likely continue to be addressed)

cameronc137 commented on gdmlGeneratorV1_materials.pl:
```diff
@@ -830,7 +833,7 @@
         </opticalsurface>";
 print def "
-        <opticalsurface name=\"Mylar\" model=\"glisur\" finish=\"polishlumirrorair\" type=\"dielectric_metal\" value=\"1.0\">
+        <opticalsurface name=\"Mylar\" model=\"glisur\" finish=\"polishedlumirrorair\" type=\"dielectric_metal\" value=\"1.0\">
```
This was a legitimate bug, but I don't think the polish properties have a large impact on remoll outputs (which may be its own problem)

cameronc137 commented on materialsOptical.xml:
```diff
@@ -66,9 +115,8 @@
 </material>
 
 <material Z="19" name="Photocathode" state="solid">
-      <property name="EFFICIENCY" ref="Cathode_Surf_Efficiency"/>
```
PMT efficiency shows up in the matrices.xml output files where a skin-surface volume is created, not in the bulk-material properties

cameronc137 commented:
Generating a new ring 5 open sector, it appears that the open sector 1 is defined as the open sector immediately next to the closed sector that is defined as the +x axis residing closed sector (i.e. the magnet coils rest on +x axis, pictured below is the electron flux, where the gap is the +x axis "coil 1" and the -x axis is "open sector 4")

![image](https://user-images.githubusercontent.com/12798552/86699064-dc203f80-bfdd-11ea-90bf-2c2bf8c6eb99.png)

So, to match this coordinate frame we should have sector 4 reside on the -x axis, but it is on the positive x axis instead. This is done by the rotation in line 1244 of the gdml generator (as it stands at this commit). This number is generated in cadGeneratorV1.pl, probably around line 115, though I can't tell the difference in logic from the older version of the code

cameronc137 committed:
Updating numbering one more time. Also updating cadp.csv
for the new shortened experiment design first-guess from Sakib
- Verified 16.5 degrees centers the flux on the PMT for ring 5.0

cameronc137 committed:
Adding in a new parallel world detector generator
- simple x-y plane projections of the Quartz, reflector, light guide,
  and PMT

cameronc137 committed:
Updated the gdml generator to get the rotations and shape of the light
guide correctly. Note I've only tested this for ring 5 open, so there
may be many more issues with rotations.

cameronc137 committed:
Updating the rotations to put ring 5-open on the -x axis and stopping
relying on position placements to get the angles right (just use angles)

cameronc137 committed:
Updating flat-idealized detectors to have exactly encompassing parent holding volumes

Remaining task: make the encompassing cube mother volume intelligently encompass only the detectors in question

cameronc137 committed:
Updating the numbering scheme detector selection to do the following:
- If no detector number is given automatically print all detectors in all rings
- If a detector segment is given but no number, automatically print all rings of that segment, including all sub-segments of ring 5s
- If a detector segment and ring number is given then print just that one detector
- If that detector is in ring 5 then only print the central sub-segment detector
- These all work with overloaded segment names (i.e. "opentransclosed" does what you would want it to do)

Updating the parallel world flat detectors to sit in the right spot exactly

cameronc137 committed:
Adding a new version of the gdml generator, this time that will only place 1/100th of the reflector tile, but with theta and phi rotations applied to it (using the strange structure of the xtrusions already in place)

cameronc137 committed:
Updating gdmlGenerator and materialsOptical to match Qsim implementation

wdconinc commented:
What is the status of this PR? What are the issues that still need to be resolved?

wdconinc commented:
@cameronc137 @rahmans1 Status update? Does not merit merging?

cameronc137 commented:
I think this one is still being worked on in another branch. I’ve included the new numbering in my recent analyses. I need to tie that analysis off soon. Sorry for the wait

cameronc137 committed:
Updating cadp.csv tracking so the most recent blessed detector design is cadp.csv
The current blessed cadp.csv was made with the shortened MOLLER hybrid magnetic fields (which have been superceded since then)

cameronc137 committed:
Putting the reflector sub-segmentation steps into the main gdml generator.
Some improvements can still be made to the subsegmentation, including more control over segment size and of course the shape implementation (currently just makes a rectangle out of smaller rectangles... why not parabaloids or something?)

cameronc137 committed:
Merge pull request #14 from JeffersonLab/feature-auto-parallel

Adding in a new parallel world detector generator

cameronc137 commented:
As a result of the work in feature-auto-parallel (Feb 7th), this branch is also finished and can be merged into main branch.
